packages/rollup-plugin-import-meta-assets/test/snapshots/dynamic-vars.js

```diff
@@ -0,0 +1,21 @@
+function __variableDynamicURLRuntime0__(path) {
+  switch (path) {
+    case './dynamic-assets/one.svg': return new URL(new URL('assets/one-824f522a.svg', import.meta.url).href, import.meta.url);
```

Comment:
Yes. They are needed here, but not needed in the original code, which was rewritten. Removing it from the rewritten original code is tricky because of spaces.

Reply:
the outside URL wrapper is not needed, is it?

Reply:
Ah, I see what you mean. The doubling up comes from the process being performed in two parts. Not sure if we can prevent that.

Reply:
We need to wrap it in `new URL` so that the second part also parses it and adds the hash.

Reply:
not sure if that's a good enough example, but

```
> new URL(new URL('assets/one-824f522a.svg', 'http://localhost/some-path/').href, 'http://localhost/some-path/').href
'http://localhost/some-path/assets/one-824f522a.svg'
> new URL('assets/one-824f522a.svg', 'http://localhost/some-path/').href
'http://localhost/some-path/assets/one-824f522a.svg'
```

Reply:
I'm missing smth here
- how does the 2nd one add the hash?
- parsing is just parsing, without generating code it won't change much from the functional perspective?

Reply:
This behavior of an extra wrapping `new URL()` is not something that we added in this PR, it happens for every snapshot in the test if you look at them. 

It's probably required for when you have this code:

```js
const searchParams = new URL('./four.svg', import.meta.url).searchParams;
```

and that becomes 

```js
const searchParams = new URL(new URL('assets/four-97848687.svg', import.meta.url).href, import.meta.url).searchParams;
```

which is not the same as

```js
const searchParams = new URL('assets/four-97848687.svg', import.meta.url).searchParams;
```
So the wrapping new URL() + href prop is needed then?